internal/testing/util_test.go

```diff
@@ -1,36 +1,54 @@
 package testing
 
 import (
+	"reflect"
 	"testing"
 
 	"github.com/gophercloud/gophercloud/internal"
-	th "github.com/gophercloud/gophercloud/testhelper"
 )
 
 func TestRemainingKeys(t *testing.T) {
 	type User struct {
-		FirstName string `json:"first_name"`
-		LastName  string `json:"last_name"`
-		City      string
+		UserID    string `json:"user_id"`
+		Username  string `json:"username"`
+		Location  string `json:"-"`
+		CreatedAt string `json:"-"`
+		Status    string
+		IsAdmin   bool
 	}
 
-	userStruct := User{
-		FirstName: "John",
-		LastName:  "Doe",
+	userResponse := map[string]interface{}{
+		"user_id":      "abcd1234",
+		"username":     "jdoe",
+		"location":     "Hawaii",
+		"created_at":   "2017-06-08T02:49:03.000000",
+		"status":       "active",
+		"is_admin":     "true",
+		"custom_field": "foo",
 	}
 
-	userMap := map[string]interface{}{
-		"first_name": "John",
-		"last_name":  "Doe",
-		"city":       "Honolulu",
-		"state":      "Hawaii",
+	expected := map[string]interface{}{
+		"created_at":   "2017-06-08T02:49:03.000000",
+		"is_admin":     "true",
+		"custom_field": "foo",
 	}
 
-	expected := map[string]interface{}{
-		"city":  "Honolulu",
-		"state": "Hawaii",
+	notExpected := map[string]interface{}{
+		"created_at":   "2017-06-08T02:49:03.000000",
+		"is_admin":     "true",
+		"custom_field": "foo",
+		"bar":          "baz",
+	}
+
+	actual := internal.RemainingKeys(User{}, userResponse)
+
+	isEqual := reflect.DeepEqual(expected, actual)
+	if !isEqual {
+		t.Fatalf("expected %s but got %s", expected, actual)
 	}
 
-	actual := internal.RemainingKeys(userStruct, userMap)
-	th.AssertDeepEquals(t, expected, actual)
+	isEqual = reflect.DeepEqual(notExpected, actual)
+	if isEqual {
+		t.Fatalf("expected %s but got %s", notExpected, actual)
+	}
```

Comment:
This could technically be removed. It was more of a sanity check from the `th.AssertDeepEquals` bug.